libafl/src/events/tcp.rs

```diff
@@ -194,17 +253,26 @@ where
                         let len_buf: [u8; 4] = len.to_le_bytes();
 
                         // Write message length
-                        write.write_all(&len_buf).await.expect("Writing failed");
+                        if write.write_all(&len_buf).await.is_err() {
+                            // The socket is closed, the client is restarting
+                            log::info!("Socket closed, client restarting");
+                            return;
+                        }
                         // Write the rest
-                        write.write_all(&buf).await.expect("Socket closed?");
+                        if write.write_all(&buf).await.is_err() {
+                            // The socket is closed, the client is restarting
+                            log::info!("Socket closed, client restarting");
+                            return;
+                        }
                     }
                 });
             }
-            println!("joining handles..");
+
+            /*log::info!("Joining handles..");
```

Comment:
Maybe remove instead of comment out?

Reply:
it will be restored after having await_restart_safe